src/utils/helpers.ts

```diff
@@ -0,0 +1,18 @@
+export const buildPath = (...args: string[]): string => {
+  return args
+    .map((part, i) => {
+      if (i === 0) {
+        return part.trim().replace(/[/]*$/g, '');
+      } else {
+        return part.trim().replace(/(^[/]*|[/]*$)/g, '');
+      }
+    })
+    .filter((x) => x.length)
+    .join('/');
+};
+
+export const buildRelativePath = (...args: string[]): string => {
+  const path = buildPath(...args);
+
+  return path.startsWith('/') ? path : `/${path}`;
+};
```

Comment:
The reason I did this is because of how the `react-router-dom` v5 package (the one we use) behaves with nested routes. This is how I'd assume we'd like paths to be implemented by somebody creating a page:
```
<Chapter path="/chapter-one">
    <Page path="/page-one"></Page>
</Chapter>
```

In `react-router-dom` v6, the logic inside of the `Chapter` and `Page` components wouldn't need any extra logic because the React router would understand that the `Page` is nested inside the `Chapter`, and it could render it with no issue. 

However, we are using v5 of the React router. In this version, our Page component can only render as a nested route if the `Page` internally provides its `<Route>` component with the entire relative path. This means that we must get the parent's path (the Chapter's) and append it with the child's path (the Page's). 

Initially to do this, I did try to use a template literal and concatenate the strings. However, the issue that kept happening was that there were inconsistencies across tests/example pages/etc for how the `path` prop was passed to a value. There were different permutations of paths using `/` characters at both the beginning and end, only the beginning, only the end, or not at all. This meant that when concatenating two paths together, you could get combination of URLs like so:
`/chapter-one/page-one`
`chapter-onepage-one`
`chapter-one/page-one`
`/chapter-onepage-one`
`/chapter-one//page-one`

If I remember correctly, only the first of those is correct. If any of the others were provided, we would render nothing, and no errors would output. This lead to a lot of confusion until I realized what was going on, and that's why I thought a helper function to combine paths would make sense, since we could ensure the relative URL we construct inside a `Path` is always consistent.

Reply:
Makes sense! Thanks for the explanation.

It's probably outside the scope for this ticket, but I think we should eventually look at upgrading to React Router V6 to reduce our reliance on something we know can be handled by the router. I'll create a ticket 👍